File: lib/ui/window/platform_message.cc
// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "flutter/lib/ui/window/platform_message.h"

#include <utility>

namespace blink {

PlatformMessage::PlatformMessage(std::string channel,
                                 std::vector<uint8_t> data,
                                 ftl::RefPtr<PlatformMessageResponse> response)
    : channel_(std::move(channel)),
      data_(std::move(data)),
      response_(std::move(response)) {}

PlatformMessage::~PlatformMessage() = default;

}  // namespace blink

File: engine/src/flutter/lib/web_ui/lib/src/engine/compositor/platform_message.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// @dart = 2.6
part of engine;

class PlatformMessage {
  final String channel;
  final ByteData data;
  final PlatformMessageResponse response;

  PlatformMessage(this.channel, this.data, this.response);
}

class PlatformMessageResponse {
  void complete(Uint8List data) {}
  void completeEmpty() {}
}
